### PR TITLE
finished vision, started on translate & lang

### PR DESCRIPTION
**Pull request by munkhuushmgl**

### CODE REVIEW

File: google/cloud/automl/v1beta1/automl_gapic.yaml [hunk 1]
```diff
@@ -347,6 +402,8 @@ interfaces:
       name: dataset
     resource_name_treatment: STATIC_TYPES
     timeout_millis: 5000
+   # This is also broken
```
Comment (beccasaurus) [inline google/cloud/automl/v1beta1/automl_gapic.yaml]: What's broken? :)

File: google/cloud/automl/v1beta1/automl_gapic.yaml [hunk 2]
```diff
@@ -496,11 +651,14 @@ interfaces:
       poll_delay_multiplier: 1.5
       max_poll_delay_millis: 5000
       total_poll_timeout_millis: 300000
+      #----------_> this is one messaed up
```
Comment (beccasaurus) [inline google/cloud/automl/v1beta1/automl_gapic.yaml]: Which one and what's messed up about it? :)

File: google/cloud/automl/v1beta1/automl_gapic.yaml [hunk 3]
```diff
@@ -590,6 +769,39 @@ interfaces:
...
+        - parameter: parent%project
+          sample_argument_name: project_id
+          description: Your Google Cloud Project ID
+        - parameter: parent%location
+          sample_argument_name: location
+          description: Location, e.g. us-central1.
+        - parameter: model.display_name
+          sample_argument_name: model_name
+          description: the Name of the model.
+          #-------->GETTING ERROR HERE!!!!
```
Comment (beccasaurus) [inline google/cloud/automl/v1beta1/automl_gapic.yaml:798]: Did we figure this one out? Looks familiar.

File: google/cloud/automl/v1beta1/automl_gapic.yaml (hunk 1 above)
Comment (munkhuushmgl) [reply google/cloud/automl/v1beta1/automl_gapic.yaml]: They are all fixed , ignore all these broken comments.


Han helped me to fix them, just forgot to delete them afterwards.

File: google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+test:
+  suites:
+  - name: Cloud AutoML Vision
+    setup:
+      # Read Project ID from Environment Variable
+    - code: project_id="java-docs-samples-testing"
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml]:
```yaml
setup:
- env:
    name: GOOGLE_PROJECT_ID
    variable: project_id
```

File: google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    cases:
+    - name: List Datasets
+      spec:
+      - call: 
+          sample: automl_vision_list_datasets
+          params: 
+            location: { literal: "us-central1" }
+            project_id: { variable: project_id }
+            filter: { literal: "" }
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml]: This list datasets really need an empty filter? Can this be the default which the sample uses?

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+test:
+  suites:
+  - name: Cloud AutoML Vision Model API IT
+    setup:
+      # Read Project ID from Environment Variable
+    - code: project_id="java-docs-samples-testing"
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml]: Read model & project ID from environment variables (see other comment)

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      spec:
+      - call: 
+          sample: automl_vision_get_model
+          params: 
+            location: { variable: cloud_region }
+            project_id: { variable: project_id }
+            model_id: { variable: model_id }
+      - assert_contains:
+        - literal: "Model display name:"
+        - literal: "do_not_delete_me_20190403112237"
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml]: Maybe also read this from an environment variable?

Environment variables for the tests:
```
GOOGLE_PROJECT_ID
MODEL_ID
MODEL_DISPLAY_NAME
```

File: google/cloud/automl/v1beta1/automl_gapic.yaml (hunk 1 above)
Comment (beccasaurus) [reply google/cloud/automl/v1beta1/automl_gapic.yaml]: Remove the comments? :P

File: google/cloud/automl/v1beta1/automl_gapic.yaml (hunk 3 above)
Comment (munkhuushmgl) [reply google/cloud/automl/v1beta1/automl_gapic.yaml:798]:
```/Users/munkhbayarg/apis/googleapis/artman-genfiles/java/gapic-google-cloud-automl-v1beta1/samples/src/main/java/com/google/cloud/examples/automl/v1beta1/AutomlVisionCreateModel.java:111:21: error: Syntax error, insert ":: IdentifierOrNew" to complete ReferenceExpression
/Users/munkhbayarg/apis/googleapis/artman-genfiles/java/gapic-google-cloud-automl-v1beta1/samples/src/main/java/com/google/cloud/examples/automl/v1beta1/AutomlVisionCreateModel.java:111:21: error: Syntax error, insert "AssignmentOperator Expression" to complete Assignment
/Users/munkhbayarg/apis/googleapis/artman-genfiles/java/gapic-google-cloud-automl-v1beta1/samples/src/main/java/com/google/cloud/examples/automl/v1beta1/AutomlVisionCreateModel.java:111:21: error: Syntax error, insert ";" to complete Statement

artman> Traceback (most recent call last):
  File "/artman/artman/cli/main.py", line 71, in main
    engine.run()
  File "/usr/local/lib/python3.5/dist-packages/taskflow/engines/action_engine/engine.py", line 159, in run
    for _state in self.run_iter():
  File "/usr/local/lib/python3.5/dist-packages/taskflow/engines/action_engine/engine.py", line 223, in run_iter
    failure.Failure.reraise_if_any(it)
  File "/usr/local/lib/python3.5/dist-packages/taskflow/types/failure.py", line 292, in reraise_if_any
    failures[0].reraise()
  File "/usr/local/lib/python3.5/dist-packages/taskflow/types/failure.py", line 299, in reraise
    six.reraise(*self._exc_info)
  File "/usr/local/lib/python3.5/dist-packages/six.py", line 693, in reraise
    raise value
  File "/usr/local/lib/python3.5/dist-packages/taskflow/engines/action_engine/executor.py", line 82, in _execute_task
    result = task.execute(**arguments)
  File "/artman/artman/tasks/format_tasks.py", line 39, in execute
    ['java', '-jar', path, '--replace'] + targetFiles)
  File "/artman/artman/tasks/task_base.py", line 64, in exec_command
    raise e
  File "/artman/artman/tasks/task_base.py", line 56, in exec_command
    output = subprocess.check_output(args, stderr=subprocess.STDOUT)
  File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)```

Comment (munkhuushmgl) [reply google/cloud/automl/v1beta1/automl_gapic.yaml:798]: That's what I'm getting when I uncomment out commented out section

File: google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml (hunk 5 above)
Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_tests.yaml]: Nope, it was my fault, there was posed to be "imageClassificationDatasetMetadata:*" for filter, I will add that.

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml (hunk 6 above)
Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml]: But for unit testing, are we supposed to use this project?

File: google/cloud/automl/v1beta1/automl_gapic.yaml (hunk 3 above)
Comment (yihanzhen) [reply google/cloud/automl/v1beta1/automl_gapic.yaml:798]: Oh I see. This field is an `int64` type and is generated to a `long` in Java. We are not handling long types in Java yet. Fix coming soon.

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+  suites:
+  - name: Cloud AutoML Vision Model API IT
+    setup:
+    - code: cloud_region = "us-central1"
+      # Read Project ID from Environment Variable
+    - env:
+        what: GOOGLE_PROJECT_ID
+        var: project_id
+    - env:
+        what: MODEL_ID
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml]:
```yaml
- env:
    name: MODEL_ID
    variable: model_id
```

If your sample-tester uses `what`, you should update to use the latest sample-tester

```
pip install sample-tester
```

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        var: model_display_name
+    cases:
+    - name: List Model
+      spec:
+      - call: 
+          sample: automl_vision_list_models
+          params: 
+            location: { variable: cloud_region }
+            project_id: { variable: project_id }
+            filter: { literal: "" }
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:24]: Why is this here?

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+          sample: automl_vision_get_model
+          params: 
+            location: { variable: cloud_region }
+            project_id: { variable: project_id }
+            model_id: { variable: model_id }
+      - assert_contains:
+        - literal: "Model display name:"
+        - variable: model_display_name
+
+    # TODO - listModelEvaluations
```
Comment (beccasaurus) [inline google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:43]: What are these TODOs? Could you update with a more descriptive message? :)

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml (hunk 9 above)
Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:24]: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/8956f1741131ea22063466dbc3e74aa1d26144d4/vision/automl/src/test/java/com/google/cloud/vision/samples/automl/ModelApiIT.java#L55-L57

Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:24]: I thought I was converting old IT from java-docs-samples

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml (hunk 10 above)
Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:43]: I just forgot to delete them afterwards

File: google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml (hunk 9 above)
Comment (munkhuushmgl) [reply google/cloud/automl/unit-tests/cloud_automl_vision_model_tests.yaml:24]: Did you mean why there are two different unit tests files ? @beccasaurus

File: google/cloud/automl/v1beta1/automl_gapic.yaml [hunk 11]
```diff
@@ -623,6 +974,64 @@ interfaces:
...
+    
+    sample_value_sets:
+    - id: automl_vision_list_models
+      description: Demonstrates using the AutoML client to get model details.
+
+      parameters:
+        defaults:
+        - parent%project="[Google Cloud Project ID]"
+        - parent%location="us-central1"
+        - filter="default_filter"
```
Comment (beccasaurus) [inline google/cloud/automl/v1beta1/automl_gapic.yaml]: Is `"default_filter"` a valid filter?

In the tests, you pass `""`. Should the default perhaps be an empty string?

I would try updating this to be an empty string :)
